sdks/rust/golem-rust-macro/src/agentic/agent_implementation_impl.rs

```diff
@@ -157,7 +157,17 @@ fn generate_method_param_extraction(param_idents: &[syn::Ident]) -> Vec<proc_mac
         quote! {
             let element_value_result = match &input {
                 golem_rust::golem_agentic::golem::agent::common::DataValue::Tuple(values) => {
-                    values.get(#i).expect("missing argument").clone()
+                    let value = values.get(#i);
+
+                    match value {
+                        Some(v) => Ok(v.clone()),
+                        None => Err(golem_rust::golem_agentic::golem::agent::common::AgentError::CustomError(
+                            golem_rust::wasm_rpc::ValueAndType::new(
+                                golem_rust::wasm_rpc::Value::String(format!("Missing arguments at pos {}", #i)),
+                                golem_rust::wasm_rpc::analysis::analysed_type::str(),
+                            ).into(),
+                        )),
```

Comment:
There was a bug that the macro expansion didn't compile for the integration test